### PR TITLE
isisometric for negative definite Zlattices

### PR DESCRIPTION
**Pull request by simonbrandhorst**

### CODE REVIEW

File: test/QuadForm/Quad/ZLattices.jl [hunk 1]
```diff
@@ -188,8 +188,11 @@ end
   s = sprint(show, "text/plain", Lr0)
   @test occursin("lattice", s)
 
-  # automorphisms
+  # isometry testing
   C1 = root_lattice(:A, 2)
+  C1m = twist(C1,-1)
+  @test isisometric(C1,C1)[1]
```
Comment (simonbrandhorst) [inline test/QuadForm/Quad/ZLattices.jl:194]:
```suggestion
  @test isisometric(C1m, C1m)[1]
```

File: test/QuadForm/Quad/ZLattices.jl [hunk 2]
```diff
@@ -188,8 +188,11 @@ end
   s = sprint(show, "text/plain", Lr0)
   @test occursin("lattice", s)
 
-  # automorphisms
+  # isometry testing
   C1 = root_lattice(:A, 2)
+  C1m = twist(C1,-1)
```
Comment (simonbrandhorst) [inline test/QuadForm/Quad/ZLattices.jl:193]:
```suggestion
  C1m = rescale(C1,-1)
```

File: src/QuadForm/Quad/ZLattices.jl [hunk 3]
```diff
@@ -340,6 +340,17 @@ end
 # documented in ../Lattices.jl
 
 function isisometric(L::ZLat, M::ZLat; ambient_representation::Bool = true)
+  @req rank(L) == 0 || isdefinite(L) && isdefinite(M) "The lattices must be definite"
+
+  if rank(L) == 0 && rank(M) == 0
+    return true
```
Comment (thofma) [inline src/QuadForm/Quad/ZLattices.jl:346]:
```suggestion
    return true, identity_matrix(ZZ, 0)
```

Comment (thofma) [reply src/QuadForm/Quad/ZLattices.jl:346]: Also, is the rank zero lattice not definite?

**Pull request merged**